### PR TITLE
[Refactor:TAGrading] Better Query Logic For Gradeable Average

### PR DESCRIPTION
**Pull request by Youssef-Chip**
### Please check if the PR fulfills these requirements:

* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant
* [ ] Screenshots are attached to Github PR if visual/UI changes were made

### What is the current behavior?
A query responsible for including/omitting bad submissions could be written better.
### What is the new behavior?
Rewritten query in a better format

### CODE REVIEW

**Comment by williamjallen** [comment]
Please add a bit more info to your PR title so it's possible to understand roughly what this PR is about without looking at the PR body or code changes.

**Review by ziesski — APPROVED** [review]
Looks good, reduce chunky code

File: site/app/libraries/database/DatabaseQueries.php [hunk 1]
```diff
@@ -2497,13 +2497,12 @@ public function getAverageForGradeable($g_id, $section_key, $is_team, string $ov
...
-            else {
-                $bad_submissions_condition = "INNER JOIN late_day_cache AS ldc ON ldc.{$user_or_team_id} = {$u_or_t}.{$user_or_team_id} AND ldc.g_id=gc.g_id AND 
-                    (ldc.submission_days_late = 0 OR ldc.late_days_change != 0)";
-            }
+            $bad_submissions_condition = "INNER JOIN(
+                SELECT DISTINCT ldc.{$user_or_team_id}
+                FROM late_day_cache AS ldc
+                WHERE ldc.g_id=? AND ( submission_days_late = 0 OR ldc.late_days_change != 0 
+              ) )AS ldc ON ldc.{$user_or_team_id}={$u_or_t}.{$user_or_team_id}";
+            $params = [$g_id,$g_id, $count];
```
Comment (ziesski) [inline site/app/libraries/database/DatabaseQueries.php:2505]: why are we having multiple $g_id params?

**Review by ziesski — CHANGES_REQUESTED** [review]

File: site/app/libraries/database/DatabaseQueries.php (hunk 1 above)
Comment (Youssef-Chip) [reply site/app/libraries/database/DatabaseQueries.php:2505]: The code I added required one more extra parameter (one extra g_id), if you go to line 2468, it has one less parameter. So only when the filter bad submission is omitted, it uses one extra parameter than usual.

File: site/app/libraries/database/DatabaseQueries.php [hunk 2]
```diff
@@ -2497,13 +2497,12 @@ public function getAverageForGradeable($g_id, $section_key, $is_team, string $ov
...
-            }
-            else {
-                $bad_submissions_condition = "INNER JOIN late_day_cache AS ldc ON ldc.{$user_or_team_id} = {$u_or_t}.{$user_or_team_id} AND ldc.g_id=gc.g_id AND 
-                    (ldc.submission_days_late = 0 OR ldc.late_days_change != 0)";
-            }
+            $bad_submissions_condition = "INNER JOIN(
+                SELECT DISTINCT ldc.{$user_or_team_id}
+                FROM late_day_cache AS ldc
+                WHERE ldc.g_id=? AND ( submission_days_late = 0 OR ldc.late_days_change != 0 
+              ) )AS ldc ON ldc.{$user_or_team_id}={$u_or_t}.{$user_or_team_id}";
```
Comment (ziesski) [inline site/app/libraries/database/DatabaseQueries.php:2504]: check line 2494 how it adds [$g_id]

**Review by ziesski — APPROVED** [review]
looks good :)

**Pull request merged**